shared/src/commonMain/kotlin/vn/luongvo/kmm/survey/domain/usecase/GetUserProfileUseCase.kt

```diff
@@ -0,0 +1,17 @@
+package vn.luongvo.kmm.survey.domain.usecase
+
+import kotlinx.coroutines.flow.Flow
+import vn.luongvo.kmm.survey.domain.model.User
+import vn.luongvo.kmm.survey.domain.repository.UserRepository
+
+interface GetUserProfileUseCase {
+
+    operator fun invoke(): Flow<User>
+}
+
+class GetUserProfileUseCaseImpl(private val repository: UserRepository) : GetUserProfileUseCase {
```

Comment:
please follow our common practice
```suggestion
class GetUserProfileUseCaseImpl(private val userRepository: UserRepository) : GetUserProfileUseCase {
```

Reply:
@manh-t I see all of the comments are about having `class val` fits its class name. My questions are:
- Why do we need to clarify that when having `only 1 repository in this use case`? What's the benefit of that? That's the same as when working with a screen, we just need to know that screen has 1 `navigator` and 1 `viewModel`. Do we really need to clarify whether they are `mainNavigator` and `homeViewModel` as params or vals in the HomeScreen?
- I don't think this is a common practice. Actually, I clarify their names in case there are 2 same components like `AuthRemoteDataSource` and `TokenLocalDataSource` here.

https://github.com/luongvo/kmm-survey/blob/491066b7fd56b4f967f43bb3d8fd73f57d132ba1/shared/src/commonMain/kotlin/vn/luongvo/kmm/survey/data/repository/AuthRepositoryImpl.kt#L14-L15

FYI: another sample

![image](https://user-images.githubusercontent.com/16315358/209636193-7726411f-cda9-43b0-8942-64afa91eec2f.png)

Reply:
> Why do we need to clarify that when having only 1 repository in this use case? What's the benefit of that? That's the same as when working with a screen, we just need to know that screen has 1 navigator and 1 viewModel.

Even if there is only 1 repository, we still need to make it clear IMO.
- `repository` is not a descriptive name, please check [our compass](https://nimblehq.co/compass/development/code-conventions/kotlin/#convention-13). [A good name for a variable is usually a noun or a noun phrase explaining what the class is](https://kotlinlang.org/docs/coding-conventions.html#choose-good-names), right? 😉 

> Do we really need to clarify whether they are mainNavigator and homeViewModel as params or vals in the HomeScreen?

Yes, I think we need to clarify them too. 🙇 

Regarding the example image above, the variables `border`, `colors` are members of the class `Button`, so they can be understood as the button's border, am I correct? 🙏  And they are also a common term for Compose, and Flutter, so I'm not against it

Reply:
@manh-t 
> repository is not a descriptive name, please check [our compass](https://nimblehq.co/compass/development/code-conventions/kotlin/#convention-13). [A good name for a variable is usually a noun or a noun phrase explaining what the class is](https://kotlinlang.org/docs/coding-conventions.html#choose-good-names), right? 😉

Yes, you pointed to a correct [convention](https://kotlinlang.org/docs/coding-conventions.html#formatting), however, it applies for naming a class or method, not a val or param. Actually, I named my classes or methods so clearly: `TokenLocalDataSource` or `UserRepository`.

Let me show you some other [examples](https://github.com/chrisbanes/tivi/blob/main/domain/src/main/java/app/tivi/domain/interactors/ClearUserDetails.kt#L27). Yes, we should name our vals and params clear but we should not force to apply it in all places:
- I kept using full name use case names in VM https://github.com/luongvo/kmm-survey/blob/491066b7fd56b4f967f43bb3d8fd73f57d132ba1/android/src/main/java/vn/luongvo/kmm/survey/android/ui/screens/login/LoginViewModel.kt#L11-L12.
- For use case, normally, 1 use case attached by a repository so I don't think we need the full repository name here https://github.com/luongvo/kmm-survey/blob/491066b7fd56b4f967f43bb3d8fd73f57d132ba1/shared/src/commonMain/kotlin/vn/luongvo/kmm/survey/domain/usecase/LogInUseCase.kt#L13-L19

IMO, this's more about personal-view with no convention yet. I don't think it's a big issue to focus on, please 🙏

Reply:
@luongvo yes, this is about the developer's taste 🙆‍♂️ Agree to leave it as it is 🆗